test/flush_worker_test.cc

```diff
@@ -0,0 +1,149 @@
+#include <functional>
+#include <thread>
+
+#include "external/envoy/source/common/common/random_generator.h"
+#include "external/envoy/source/common/event/dispatcher_impl.h"
+#include "external/envoy/source/common/runtime/runtime_impl.h"
+#include "external/envoy/source/common/stats/isolated_store_impl.h"
+#include "external/envoy/test/mocks/api/mocks.h"
+#include "external/envoy/test/mocks/local_info/mocks.h"
+#include "external/envoy/test/mocks/protobuf/mocks.h"
+#include "external/envoy/test/mocks/stats/mocks.h"
+#include "external/envoy/test/mocks/thread_local/mocks.h"
+#include "external/envoy/test/test_common/simulated_time_system.h"
+
+#include "client/flush_worker_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace Client {
+
+using namespace testing;
+
+class FlushWorkerTest : public Test {
+public:
+  FlushWorkerTest()
+      : thread_factory_(Envoy::Thread::threadFactoryForTest()),
+        dispatcher_(new NiceMock<Envoy::Event::MockDispatcher>()) {
+    loader_ = std::make_unique<Envoy::Runtime::ScopedLoaderSingleton>(
+        Envoy::Runtime::LoaderPtr{new Envoy::Runtime::LoaderImpl(
+            *dispatcher_, tls_, {}, local_info_, store_, rand_, validation_visitor_, mock_api_)});
+    sink_ = new StrictMock<Envoy::Stats::MockSink>();
```

Comment:
Two questions here:
1.  It appears that sink_ is a pointer that we are then passing into stats_sinks as a unique_ptr.  That doesn't seem like it should work.
2.  Why does sink_ need to be a member variable of FlushWorkerTest?  It is already exposed as part of stats_sinks_ being a unique pointer to it.

Reply:
1. I think it can work because the unique_ptr has a constructor to take a raw pointer as argument. 
Same e.g. in Envoy https://github.com/envoyproxy/envoy/blob/cf850688a93b8dea5705170a211ea98b77beda61/test/server/server_test.cc#L66-L67

2. It needs to be  member variable since later in the test we check 
`EXPECT_CALL(*sink_, flush(_)).Times();`.
We can not replace it with stats_sinks_ in the `EXPECT_CALL()` since `EXPECT_CALL()` is expected to take `MockSink` instead of `Sink`